namer/namer.cc

```diff
@@ -1049,6 +1094,44 @@ class SymbolDefiner {
         }
     }
 
+    core::FieldRef insertField(core::MutableContext ctx, const core::FoundField &field) {
+        core::ClassOrModuleRef scope;
+
+        // resolver checks a whole bunch of various error conditions here; we just want to
+        // know where to define the field.
+        if (field.kind == core::FoundField::Kind::Class) {
+            scope = ctx.owner.enclosingClass(ctx);
+        } else {
+            scope = ctx.selfClass();
+        }
+
+        auto existing = scope.data(ctx)->findMember(ctx, field.name);
+        if (existing.exists() && existing.isFieldOrStaticField()) {
+            auto prior = existing.asFieldRef();
+            // There was a previous declaration of this field in this file.  Ignore it;
+            // let resolver deal with issuing the appropriate errors.
+            if (prior.data(ctx)->resultType == core::Types::todo()) {
+                return prior;
+            }
+
+            // We are on the fast path and there was a previous declaration.
+            // TODO(froydnj) apparently incremental resolve can't add new symbols?!
+            // ctx.state.mangleRenameSymbol(existing, existing.data(ctx)->name);
+            prior.data(ctx)->resultType = core::Types::todo();
```

Comment:
Out of curiosity, did you play around with setting/not setting `resultType` to `todo()` in this case?

For example I don't think that we reset a method's type to some initial value when we discover a pre-existing method on the fast path.

I don't have a strong opinion on it either way (especially because actual changes to the types of fields currently takes the slow path anyways) but I was curious to know whether you discovered that it mattered.